### ISSUE TITLE
FormattedMessagePlaceholderValues don't get serialized properly

### ISSUE THREAD

**Issue by gkfischer** [benmccallum/fairybread#33]
The dictionary FormattedMessagePlaceholderValues doesn't get serialized properly in the Failures extension in the GraphQL response for me. It only turns out as 
```
FormattedMessagePlaceholderValues = System.Collections.Generic.Dictionary`2[System.String,System.Object]
````

Any ideas how to fix that would be highly appreciated.

Version: 3.x

**Comment by benmccallum** [comment]
Thanks for logging @gkfischer , I'll have a look this week/weekend. 

I think it's a problem in the error handler. It seems like HC's serialization out of the server doesn't like to properly handle collections like this. I had a similar problem recently that I solved by `.ToArray()`ing an IEnumerable property, so I guess we'll have to do a similar thing here to make it something HC will serialize out. 

I need to add integration tests that utilise the HC serializer it seems, as this is the second issue with it that was missed.

**Comment by benmccallum** [comment]
This might be resolved in `4.0.0-preview.1`, need to test.

**Comment by gkfischer** [comment]
Yes it's working in 4.0.0-preview.1 properly. I get somthing like that as reponse in case of an validation error

```
"errors": [
    {
      "message": "A component with name 'Test Component' and the same type already exists.",
      "path": [
        "addComponent"
      ],
      "extensions": {
        "code": "FairyBread_ValidationError",
        "errorCode": "ComponentNameValidator",
        "errorMessage": "A component with name 'Test Component' and the same type already exists.",
        "propertyName": "Name",
        "attemptedValue": "Test Component",
        "severity": "Error",
        "formattedMessagePlaceholderValues": {
          "PropertyName": "Name",
          "PropertyValue": "Test Component"
        }
      }
    }
  ],
```

Thanks
